devito/dle/backends/advanced.py

```diff
@@ -162,7 +162,7 @@ def _loop_blocking(self, state, **kwargs):
                     # Build Iteration over blocks
                     dim = blocked.setdefault(i, Dimension("%s_block" % i.dim.name))
                     block_size = dim.symbolic_size
-                    iter_size = i.dim.size or i.dim.symbolic_size
+                    iter_size = i.dim.size if i.dim.is_Fixed else i.dim.symbolic_size
```

Comment:
At this point I think having just ``i.dim.symbolic_size`` (i.e., with the ``if ... else``) should work. Could you try this?

Reply:
Do you mean without the `if...else`?